src/plugins/vis_type_timeseries/public/application/components/vis_types/table/vis.js

```diff
@@ -57,27 +60,44 @@ class TableVis extends Component {
   }
 
   renderRow = (row) => {
-    const { model } = this.props;
+    const { model, fieldFormatMap, getConfig } = this.props;
 
     let rowDisplay = getValueOrEmpty(
       model.pivot_type === 'date' ? this.dateFormatter.convert(row.key) : row.key
     );
 
+    // we should skip url field formatting for key if tsvb have drilldown_url
+    if (fieldFormatMap?.[model.pivot_id]?.id !== FIELD_FORMAT_IDS.URL || !model.drilldown_url) {
+      const formatter = createFieldFormatter(model?.pivot_id, fieldFormatMap, 'html');
+      rowDisplay = <span dangerouslySetInnerHTML={{ __html: formatter(rowDisplay) }} />; // eslint-disable-line react/no-danger
+    }
+
     if (model.drilldown_url) {
       const url = replaceVars(model.drilldown_url, {}, { key: row.key });
       rowDisplay = <a href={sanitizeUrl(url)}>{rowDisplay}</a>;
     }
+
     const columns = row.series
       .filter((item) => item)
       .map((item) => {
         const column = this.visibleSeries.find((c) => c.id === item.id);
         if (!column) return null;
-        const formatter = createTickFormatter(
-          column.formatter,
-          column.value_template,
-          this.props.getConfig
+        const isFieldFormatter = column.formatter === DATA_FORMATTERS.DEFAULT;
+        const field = getMetricsField(column.metrics);
+        const formatter = isFieldFormatter
+          ? createFieldFormatter(field, fieldFormatMap, 'html')
+          : createTickFormatter(column.formatter, column.value_template, getConfig);
```

Comment:
nitpick: this formula gets repeated over and over in the PR. Maybe it can be isolated in a separated function that returns `{isFieldFormatter, formatter}`?

Reply:
updated `createFieldFormatter` function, could you review that part once again, please?